cron/alerts/main.go

```diff
@@ -0,0 +1,330 @@
+	return region
+}
+
+func splitStringList(compoundValue string) []string {
+	initialList := strings.Split(compoundValue, EnvListDelimiter)
+	output := []string{}
+	for _, s := range initialList {
+		output = append(output, strings.Trim(s, " "))
+	}
+	return output
+}
+
+func getZoneNames(a *AlertsConfig) error {
+	gluedZoneNames := os.Getenv(EnvKeyCFZoneNames)
+	if gluedZoneNames == "" {
+		return fmt.Errorf("required value missing for environment variable %s", EnvKeyCFZoneNames)
+	}
+	a.CFZoneNames = splitStringList(gluedZoneNames)
+
+	return nil
+}
+
+func getRecipientAddresses(a *AlertsConfig) error {
+	gluedRecipients := os.Getenv(EnvKeySESRecipients)
+	if gluedRecipients == "" {
+		return fmt.Errorf("required value missing for environment variable %s", EnvKeySESRecipients)
+	}
+
+	a.RecipientEmails = splitStringList(gluedRecipients)
+	return nil
+}
+
+func getCFContainsStrings(a *AlertsConfig) error {
+	gluedSearchStrings := os.Getenv(EnvKeyCFContainsStrings)
+	if gluedSearchStrings == "" {
+		return fmt.Errorf("required value missing for environment variable %s", EnvKeyCFContainsStrings)
+	}
+
+	a.CFContainsStrings = splitStringList(gluedSearchStrings)
+	return nil
+}
+
+func getRequiredString(envKey string, configEntry *string) error {
+	if *configEntry != "" {
+		return nil
+	}
+
+	value := os.Getenv(envKey)
+	if value == "" {
+		return fmt.Errorf("required value missing for environment variable %s", envKey)
+	}
+	*configEntry = value
+
+	return nil
+}
+
+func (a *AlertsConfig) init() error {
+
+	if err := getRequiredString(EnvKeyCFApiEmail, &a.CFApiEmail); err != nil {
+			return err
+		}
+
+	if err := getRequiredString(EnvKeyCFApiKey, &a.CFApiKey); err != nil {
+			return err
+	}
+
+	if len(a.CFContainsStrings) < 1 {
+		if err := getCFContainsStrings(a); err != nil {
+			return err
+		}
+	}
+
+	if len(a.CFZoneNames) < 1 {
+		if err := getZoneNames(a); err != nil {
+			return err
+		}
+	}
+
+	if len(a.RecipientEmails) < 1 {
+		if err := getRecipientAddresses(a); err != nil {
+			return err
+		}
+	}
+
+	if a.SESCharSet == "" {
+		a.SESCharSet = SESCharSet
+	}
+
+	if err := getRequiredString(EnvKeySESReturnToAddress, &a.SESReturnToAddr); err != nil {
+		return err
+	}
+
+	if a.SESAWSRegion == "" {
+		a.SESAWSRegion = getSESAWSRegion()
+	}
+
+	if err := getRequiredString(EnvKeySESSubject, &a.SESSubjectText); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type AlertsConfig struct {
+	CFApiKey string `json:"CFApiKey"`
+	CFApiEmail string `json:"CFApiEmail"`
+	CFZoneNames      []string `json:"CFZoneNames"`
+	CFContainsStrings []string   `json:"CFContainsString"`
+	SESAWSRegion     string   `json:"SESAWSRegion"`
+	SESCharSet       string   `json:"SESCharSet"`
+	SESReturnToAddr  string   `json:"SESReturnToAddr"`
+	SESSubjectText   string   `json:"SESSubjectText"`
+	RecipientEmails  []string `json:"RecipientEmails"`
+}
+
+func getCFRecordsWithSubstring(substring, zoneName string, recs []cloudflare.DNSRecord, results map[string][]string) {
```

Comment:
Do you need a test for `getCFRecordsWithSubstring`?

Reply:
Yep. I'll try to get back to this.